dpytools/http/dataset/base_api.py

```diff
@@ -18,10 +21,40 @@ def __init__(self, url_netloc: str, url_path: str, backoff_max: int = 30):
 
     @abstractmethod
     def post_json(self, json_data: Dict):
-        """Send a POST request with JSON data."""
-        pass
+        """
```

Comment:
You don't need the concrete implementations of `put_json` and `post_json` here - remove the body of both methods and replace with `pass` or `...`